pkg/controller/controller.go

```diff
@@ -92,17 +92,18 @@ type ErrorDetails struct {
 
 const iapHeaderName = "x-goog-authenticated-user-email"
 
-func New(h *renderer.Renderer, p *justification.Processor, allowlist []string) *Controller {
-	categories := make(map[string]struct{})
-	for v := range p.Validators() {
-		categories[v] = struct{}{}
+func New(h *renderer.Renderer, p *justification.Processor, allowlist []string) (*Controller, error) {
+	categories, err := getCatagoriesDisplayData(p.Validators())
```

Comment:
👍 good call on pre-computing this at initialization.

Reply:
Thank you.